src/sensors/temperature/TMP36.cpp

```diff
@@ -0,0 +1,17 @@
+#include "TMP36.h"
+
+using namespace wlp;
+
+TMP36::TMP36(Board::AnalogPin analog_pin)
+    : pin(analog_pin) {}
+
+double TMP36::read_temperature() {
+    pin.powerup();
```

Comment:
for this, we can follow the a convention of other sensors and have a `tmp36.begin()` function which would call `pin.powerup()` and then `tmp36.sleep()` which would call `pin.powerdown()`